### PR TITLE
Add cupy-rocm-4-0 builder

### PR DESCRIPTION
**Pull request by kmaehashi**
Run `./build-rocm.sh rocm-4.0 3.8.0` to build packages.

Blocked by https://github.com/cupy/cupy/pull/4717

### CODE REVIEW

File: builder/Dockerfile [hunk 1]
```diff
@@ -33,6 +33,11 @@ RUN ( [ ! -d /cudnn/cuda/lib64 ]   || mv /cudnn/cuda/lib64/*   /usr/local/cuda/l
     ( [ ! -d /cudnn/cuda/include ] || mv /cudnn/cuda/include/* /usr/local/cuda/include ) && \
     ldconfig
 
+# Install additional dependencies for ROCm.
+RUN [ -d /opt/rocm ] && \
+    yum -y install hipcub4.0.1 hipblas4.0.1 rocrand4.0.1 rocfft4.0.1 rocsolver4.0.1 && \
```
Comment (leofang) [inline builder/Dockerfile:38]: rccl is supported

**Review by emcastillo — APPROVED** [review]
LGTM

**Comment by emcastillo** [comment]
Jenkins, test this please

**Comment by kmaehashi** [comment]
Looking into test error.

**Comment by emcastillo** [comment]
```

00:24:53.775204 STDOUT 2118] | E: Unable to locate package hipcub |  
-- | -- | --
00:24:53.775213 STDOUT 2118] | E: Unable to locate package hipblas |  
00:24:53.775228 STDOUT 2118] | E: Unable to locate package rocrand |  
00:24:53.775237 STDOUT 2118] | E: Unable to locate package rocfft |  
00:24:53.775239 STDOUT 2118] | E: Unable to locate package rocsolver |  
00:24:53.775244 STDOUT 2118] | E: Unable to locate package rccl |  
00:24:54.031413 STDERR 2118] | The command '/bin/sh -c [ -d /opt/rocm ] \|\| (         export DEBIAN_FRONTEND=noninteractive &&         apt-get -y update &&         apt-get -y install hipcub hipblas rocrand rocfft rocsolver rccl &&         rm -rf /var/lib/apt/lists/* /var/cache/apt/archives/*     )' returned a non-zero code: 100

```
Need some PPA?

**Comment by kmaehashi** [comment]
Confirmed `kmeans` CuPy example run successfully with MI50 (gfx906), MI100 (gfx908) and Radeon RX 580 (gfx803).

**Comment by leofang** [comment]
Awesome!!!

**Pull request merged**